### DISCUSSION TITLE
Est ce que metadata.json doit être créer par `-x` ? comment inverser `-x` exactement ?

### DISCUSSION THREAD

**Discussion by lasconic** [jersou/studio-pack-generator#24] [General]
Il peut être intéressant de ne rien modifier sauf la structure d'un pack pour la rendre cohérente avec "La norme", ou d'extraire, de modifier les mp3 et de repackager.

J'ai essayé la chose suivante:

```
$ studio-pack-generator -x pack.zip
$ studio-pack-generator -miva pack-extract
```
resultat : pack-extract-1709493237166.zip

Si j'ouvre le résultat, la description du pack, le titre, la version, son image etc... ne sont pas préservés.
Pas sûr que "-extract" soit nécessaire ?  si le pack a le format de l'outil https://luckythecookie.github.io/lunii/ peut être on pourrait enlever le -XXXXXXX  pour le remplacer pendant le packaging ?

**Comment by jersou** [comment]
Salut,
comme répondu hier sur Discord, il faut lancer SPG sur le dossier "Cover node" dans le dossier "extract" pour refaire le pack, je vais voir pour éviter d'avoir ce dossier "de trop".
```
$ studio-pack-generator -x pack.zip
$ studio-pack-generator "pack.zip-extract/Cover node"
```
Je n'ai pas poussé le process "inverse" à fond, le but était surtout d'extraire les fichiers au format attendu par SPG pour corriger un pack plus facilement, il resterait la partie metadata à traiter, à voir le besoin... c'est déjà très "niche" :D comme feature.

**Reply by jersou** [reply]
je vais rajouter le metadata.json qui permettra de conserver les metadata, j'ai pu aussi supprimer le dossier superflu à la racine, mais je vais garder le format `<nom du zip>-extract`, au moins pour l'instant, car il y a l'option `--output-folder` pour changer la sortie.
Les chiffres à la fin représentent le timestamp au moment de la génération du zip, il pourrait éventuellement y avoir une option pour désactiver ce mécanisme et alors écraser le zip précédemment produit avec le même nom...

Et sinon, SPG n'a pas toutes les infos requises par "Name Generator" : pas l'age ni le nom du créateur.

**Reply by lasconic** [reply]
Ouaw, merci pour ta rapidité. Je débute avec la Lunii. Les deux packs que j'ai voulu reprendre ton outil m'a bien aidé. J'ai du repassé par studio pour remettre les fichiers MP3 modifiés. Avec cette nouvelle option, je vais pouvoir m'en passer !
